adapters/ethereum/index.js

```diff
@@ -180,6 +178,16 @@ function toEthereumChannel(channel) {
 	})
 }
 
+function getCoreContractForChannel(cfg, channel, ethProvider) {
+	const depositChainId =
+		channel.spec.depositChainId || cfg.depositAssetsToChainId[channel.depositAsset]
+	const provider =
+		ethProvider ||
+		(depositChainId && new providers.JsonRpcProvider(cfg.supportedChainIdsByRPC[depositChainId])) ||
+		getDefaultProvider(cfg.ETHEREUM_NETWORK)
```

Comment:
should check `cfg.supportedChainIdsByRPC[depositChainId]` before `new providers.JsonRpcProvider...` as at the moment `cfg.supportedChainIdsByRPC` and I'm not sure if it will throw or use default